data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OffsetManager.java

```diff
@@ -72,12 +73,29 @@ public OffsetManager(final Vertx vertx,
     this.vertx = vertx;
 
     this.consumer.partitionsRevokedHandler(partitions -> {
+      try {
+        // Async commit offsets.
+        final var commitFuture = commit(partitions);
+        // Remove revoked partitions.
+        partitions.forEach(offsetTrackers::remove);
+
+        // Wait for offsets to be actually committed.
+        commitFuture
+          .toCompletionStage()
+          .toCompletableFuture()
+          .get(1, TimeUnit.SECONDS);
+
+      } catch (final Exception ex) {
+        logger.warn("Failed to commit revoked partitions offsets {}", keyValue("partitions", partitions), ex);
+      } finally {
+        // Remove revoked partitions in any case.
+        partitions.forEach(offsetTrackers::remove);
+      }
       logPartitions("revoked", partitions);
-      partitions.forEach(offsetTrackers::remove);
     });
 
     this.consumer.partitionsAssignedHandler(partitions ->
-      partitions.forEach(tp -> logPartitions("assigned", partitions))
```

Comment:
@pierDipi wondering if on partition assignment, do we need to set this consumer back to the last committed offset if partition exists in offsettracker (i realize we've removed everything on revoke)....or the logic here is that when this consumer receives a new record, the offset tracker will get the new offset anyways?

Reply:
Yes, I'm trying to find a good design for this because we want to consume this event in multiple components so we need a pub/sub here

Reply:
On revoke, we remove only the offset trackers associated with the revoked partitions.

Reply:
Once we will handle new partitions, we will re-introduce new offset trackers as we see records